tests/Microsoft.Identity.Web.Test.Integration/DownstreamApiForAppCaeTests.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.Identity.Abstractions;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Lab.Api;
+using Microsoft.Identity.Web.Test.Common;
+using Microsoft.Identity.Web.Test.Common.Mocks;
+using Microsoft.Identity.Web.Test.Common.TestHelpers;
+using Microsoft.Identity.Web.TokenCacheProviders.InMemory;
+using Xunit;
+
+namespace Microsoft.Identity.Web.Test.Integration
+{
+    public class DownstreamApiForAppCaeTests
+    {
+        private readonly TimeSpan _delayTimeout = TimeSpan.FromMinutes(5);
+        private ServiceProvider _provider;
+        private IDownstreamApi _downstreamApi;
+        private readonly string _ccaSecret;
+        private MsalTestTokenCacheProvider _msalTestTokenCacheProvider;
+
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+        public DownstreamApiForAppCaeTests()
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+        {
+            KeyVaultSecretsProvider keyVaultSecretsProvider = new(TestConstants.BuildAutomationKeyVaultName);
+            _ccaSecret = keyVaultSecretsProvider.GetSecretByName(TestConstants.AzureADIdentityDivisionTestAgentSecret).Value;
+
+            // Need the secret before building the services
+            if (!string.IsNullOrEmpty(_ccaSecret))
+            {
+                BuildRequiredServices();
+            }
+            else
```

Comment:
nit: else not needed.

Reply:
Copy-pasted from other tests. Will refactor.